### PR TITLE
Refactored monitor configuring

### PR DESCRIPTION
**Pull request by naaando**
Decoupled logic around monitor configuration to make code easier to read

### CODE REVIEW

File: src/Interfaces/MutterDisplayConfig.vala [hunk 1]
```diff
@@ -79,6 +79,11 @@ public struct MutterReadMonitorInfo {
     public string vendor;
     public string product;
     public string serial;
+    public uint hash {
+        get {
+            return (connector+vendor+product+serial).hash ();
```
Comment (jeremypw) [inline src/Interfaces/MutterDisplayConfig.vala]: Spaces required before and after math operator.

File: src/Objects/Monitor.vala [hunk 2]
```diff
@@ -24,6 +24,12 @@ public class Display.Monitor : GLib.Object {
     public string vendor { get; set; }
     public string product { get; set; }
     public string serial { get; set; }
+    public uint hash {
+        get {
+            return (connector+vendor+product+serial).hash ();
+        }
+    }
+
```
Comment (jeremypw) [inline src/Objects/Monitor.vala]: Should not the Monitor object just have an immutable MonitorInfo struct? The vendor, product etc should not really be mutable variables - they correspond to real hardware.

File: src/Objects/MonitorManager.vala [hunk 3]
```diff
@@ -332,9 +219,91 @@ public class Display.MonitorManager : GLib.Object {
...
+        var monitor = get_monitor_by_hash (mutter_monitor.monitor.hash);
+        if (monitor == null) {
+            monitor = new Display.Monitor ();
+            monitors.add (monitor);
+        }
+
+        monitor.connector = mutter_monitor.monitor.connector;
+        monitor.vendor = mutter_monitor.monitor.vendor;
+        monitor.product = mutter_monitor.monitor.product;
+        monitor.serial = mutter_monitor.monitor.serial;
```
Comment (jeremypw) [inline src/Objects/MonitorManager.vala]: See previous comment about using an immutable struct for this info.

**Comment by jeremypw** [comment]
@naaando: You need to resolve conflicts with master before this can be reviewed again :).

**Comment by naaando** [comment]
It has been passed almost 5 months from the time I wrote this, and reading it now I don't think this solves the problem it's supposed to. And I don't have monitors right now so I won't be able to test it. I think we can close this if nobody has any objection.

**Pull request closed without merging**